spot-contracts/contracts/vaults/RolloverVault.sol

```diff
@@ -404,6 +404,58 @@ contract RolloverVault is
         return notes;
     }
 
+    /// @notice Deposits the underlying asset from {msg.sender} into the perp and vault systems,
+    ///         and mints both perp tokens and notes.
+    /// @param amount The amount tokens to be deposited into the vault.
+    /// @return The amount of perps and vault notes minted.
+    function deposit2(uint256 amount) external nonReentrant whenNotPaused returns (uint256, uint256) {
+        if (amount <= 0) {
+            revert UnacceptableDeposit();
+        }
+
+        uint256 totalSupply_ = totalSupply();
+        uint256 tvlBefore = getTVL();
+
+        // get deposit bond
+        IBondController depositBond = perp.getDepositBond();
+        BondTranches memory bt = depositBond.getTranches();
+
+        // NOTE: the vault may already have seniors which
+        //       we have to ignore for the sake of this operation.
+        uint256 seniorBalBefore = bt.tranches[0].balanceOf(address(this));
+
+        // transfer underlying in
+        underlying.safeTransferFrom(_msgSender(), address(this), amount);
```

Comment:
`perp.deposit` which comes after compares vault and spot TVL which could be potentially manipulated by this action. 

I'm rethinking if this function needs to be part of the vault or some other abstraction which sits above {perp,vault}